Sources/XCRemoteCache/Commands/Prebuild/XCPrebuild.swift

```diff
@@ -115,11 +115,22 @@ public class XCPrebuild {
             )
             let client: NetworkClient = config.disableHttpCache ? networkClient : cacheNetworkClient
             let remoteNetworkClient = RemoteNetworkClientImpl(client, urlBuilder)
-            let pathRemapper = try StringDependenciesRemapperFactory().build(
+            let envsRemapper = try StringDependenciesRemapperFactory().build(
                 orderKeys: DependenciesMapping.rewrittenEnvs,
                 envs: env,
                 customMappings: config.outOfBandMappings
             )
+            // As PrebuildContext assumes file format location and its filename (`all-product-headers.yaml`)
+            // do not fail in case of a missing headers overlay file. 
+            let overlayReader = JsonOverlayReader(
+                context.overlayHeadersPath,
+                mode: .bestEffort,
+                fileReader: fileManager
+            )
+            let overlayRemapper = try OverlayDependenciesRemapper(
+                overlayReader: overlayReader
+            )
+            let pathRemapper = DependenciesRemapperComposite([overlayRemapper, envsRemapper])
```

Comment:
`DependenciesRemapperComposite` has a logic to invert remappers when making the generic->local transformation